microsoft/botframework-solutions
# [Web Test Harness] Pass the authenticated user ID into web chat & update UI

## Description

I've updated the Web Test Harness to apply the fix Microsoft/BotFramework-WebChat#1447, which allows the test harness to pass an authenticated user's id to the bot conversation. This enables linked accounts to be demoed in web chat.

Since the Web Test solution may be used in customer demos, I've also applied some skinning to the UI matching the CSS stylesheet provided by the "F5" page in the bot templates. This, along with some additional content, should better tell the story of using the test harness.

I've also added a "Copy" button to easily copy the user id to the clipboard. This will remove some minor friction for presenters.

![image](https://user-images.githubusercontent.com/43043272/50189887-fa2f2d80-02db-11e9-85d8-252a9f5b83d9.png)
![image](https://user-images.githubusercontent.com/43043272/50189829-c05e2700-02db-11e9-85e3-0f4bb31e9c9c.png)

**Note**: This still has an issue with the Onboarding Dialog, in that the `ConversationUpdate` Activity via Direct Line uses the `Conversation.Id` as the `From.Id` property. Subsequent messages to the bot use the provided user id.

## Related Issue




#470 
#471 
#473 

## Testing Steps


Run the Web Test solution with a working bot, link your account and then navigate to the web chat tab. In that conversation if you trigger a skill requiring authentication you should skip the bot asking for you to sign on.

Ping me and I can provide a working appsettings using the BFVirtualAssistant demo environment.

## Checklist

If this contains changes that needs to be replicated between the Enterprise Template <-> Virtual Assistant
- [X] A duplicate issue is filed to track future work. - I can port Linked Accounts updates after for the second half of #471
